### ISSUE TITLE
Has the sorting on rubygems.org profiles been changed recently?

### ISSUE THREAD

**Issue by rubyFeedback** [rubygems/rubygems.org#4540]
Not sure if this is a bug or not - has the sorting algorithm been changed?

It now looks like this:

https://i.imgur.com/Y1aNmAF.png

Since a few days.

Before that it was sorted by total number of downloads, so the most downloaded
gem appeared first, on top. Instead, now it seems random where it appears. Also I got
quite a lot of unicorn errors in the last days, so not sure what is going on or if
this is related to anything.

IF the sorting was changed, could the reason be explained in the changelog or
monthly progress report? Right now I don't know why it is different; it is less
useful compared to the prior variant for sure, as my most downloaded gem now
appears in the middle rather than on top as before.

**Comment by dancristianb** [comment]
Just checked and I think this might be related to https://github.com/rubygems/rubygems.org/pull/4520/files#diff-7a47d335107fc35c2e79cc9c2b0a195c863594dd2343bcd48a4c770ac4af673cR11

Having some [data](https://github.com/rubygems/rubygems.org/blob/master/CONTRIBUTING.md#getting-the-data-dumps) in and checking the latest version of the profile page, I see the gems not ordered by download count.

<img width="1142" alt="Screenshot 2024-03-18 at 23 34 08" src="https://github.com/rubygems/rubygems.org/assets/32739319/70be8026-70d1-4695-bf12-2ec27757c354">

If I set the [Prosopite.raise](https://github.com/rubygems/rubygems.org/blob/master/config/initializers/prosopite.rb#L7) flag `false` and use the old `User.find_by_slug!(params[:id])` query in the `ProfilesController#show` action, I see the `rubygems_downloaded` scope [order](https://github.com/rubygems/rubygems.org/blob/master/app/models/user.rb#L38) correctly applied and the gems order by download count.

<img width="1215" alt="Screenshot 2024-03-18 at 23 35 01" src="https://github.com/rubygems/rubygems.org/assets/32739319/c31545c9-831e-49f7-afed-e8c427705228">

We could update [the show action](https://github.com/rubygems/rubygems.org/blob/master/app/controllers/profiles_controller.rb#L11) to use something like the attached, so we both avoid the n+1 and have the entries order.

```
@user = User.includes(rubygems_downloaded: %i[most_recent_version gem_download])
  .order("gem_downloads_rubygems.count DESC").strict_loading
  .find_by_slug!(params[:id])
```

Seems to work, I'm not sure if it's the best approach though 😀

<img width="603" alt="Screenshot 2024-03-18 at 23 50 11" src="https://github.com/rubygems/rubygems.org/assets/32739319/983a37e1-2926-48fc-9107-102361513dec">

**Comment by dancristianb** [comment]
I have this small https://github.com/rubygems/rubygems.org/pull/4544 PR covering the profile page gem order updates ready for review. Let me know what you think, thanks! 😀

**Comment by simi** [comment]
Thanks for the report @rubyFeedback, indeed this was changed by mistake and will be hopefully fixed by #4544 soon. :pray:
